### ISSUE TITLE
Compiler error in comrade_opt: AssertionError: sz == sizeof(Int)

### ISSUE THREAD

**Issue by KenzoKawamura** [ptiede/Comrade.jl#379]
# overview
- Compiler errors occur when running the `comrade_opt' function.
# code
## error location
```
xopt, sol = comrade_opt(post, LBFGS();
                        maxiters=1000, initial_params=prior_sample(rng, post))
```
in [closure imaging tutorial](https://ptiede.github.io/Comrade.jl/v0.11.0/tutorials/intermediate/ClosureImaging)
## error code
```
AssertionError: sz == sizeof(Int)
Stacktrace:
  [1] should_recurse(typ2::Any, arg_t::LLVM.IntegerType, byref::GPUCompiler.ArgumentCC, dl::LLVM.DataLayout)
    @ Enzyme.Compiler ~/.julia/packages/Enzyme/RvNgp/src/absint.jl:210
```
# environment
```
Status `/Project.toml`
  [99d987ce] Comrade v0.11.0
  [31c24e10] Distributions v0.25.113
  [7da242da] Enzyme v0.13.14
  [7f7a1694] Optimization v4.0.5
  [36348300] OptimizationOptimJL v0.4.1
  [3d61700d] Pyehtim v0.1.3
  [860ef19b] StableRNGs v1.0.2
  [b1ba175b] VLBIImagePriors v0.9.0
```
and
`Julia 1.10.6`

**Comment by ptiede** [comment]
Which version of Comrade are you using? This should be temporarily fixed on 0.11.1.

The actual issue is from a recent update to Enzyme that I'm working with the developers to fix.

**Comment by ptiede** [comment]
Ok so this is something to do with a custom rule I wrote for the closure likelihood. It is being investigated in https://github.com/EnzymeAD/Enzyme.jl/issues/2085
